### PR TITLE
integrate the tag selection dialog into the event creation screen

### PR DESCRIPTION
**Pull request by unglazedstamp**
based on #113

### CODE REVIEW

File: app/src/main/java/com/github/swent/echo/compose/event/EventTagEntry.kt [hunk 1]
```diff
@@ -1,42 +1,72 @@
...
+                    )
+                }
             }
+            OutlinedIconButton(
+                modifier = Modifier.testTag("add-tag-button"),
+                onClick = { dialogVisible = true }
+            ) {
+                val icon = Icons.Filled.Add
+                Icon(imageVector = icon, contentDescription = icon.name)
+            }
```
Comment (violoncelloCH) [inline app/src/main/java/com/github/swent/echo/compose/event/EventTagEntry.kt:59]: the button to add a new tag should also have a label next to the plus icon; at least "Add" or better "Add Tag" to make the button accessible for screen reader users:
image

File: app/src/main/java/com/github/swent/echo/compose/event/EventTagEntry.kt [hunk 2]
```diff
@@ -1,42 +1,72 @@
...
-                Text(tag.name)
-                Icon(
-                    imageVector = Icons.Filled.Clear,
-                    contentDescription =
-                        stringResource(R.string.edit_event_screen_delete_tag_button) + tag.name
-                )
+    var dialogVisible by remember { mutableStateOf(false) }
+    Column(modifier = Modifier.fillMaxWidth().padding(EVENT_PADDING_BETWEEN_INPUTS)) {
+        EventEntryName(stringResource(R.string.edit_event_screen_tags))
+        Row(modifier = Modifier.horizontalScroll(rememberScrollState())) {
```
Comment (violoncelloCH) [inline app/src/main/java/com/github/swent/echo/compose/event/EventTagEntry.kt:39]: I don't think using horizontal scroll on this makes sense from a UX perspective. It's very easy to accidentally click one of the tags when scrolling and therefore delete it. Couldn't that simply expand into multiple rows?

File: app/src/main/java/com/github/swent/echo/compose/event/EventTagEntry.kt [hunk 3]
```diff
@@ -1,42 +1,72 @@
...
+                    onClick = { onTagDeleted(tag) },
+                    modifier = Modifier.testTag("${tag.name}-tag-button")
+                ) {
+                    Text(tag.name)
+                    Icon(
+                        imageVector = Icons.Filled.Clear,
+                        contentDescription =
+                            stringResource(R.string.edit_event_screen_delete_tag_button) + tag.name
+                    )
+                }
```
Comment (violoncelloCH) [inline app/src/main/java/com/github/swent/echo/compose/event/EventTagEntry.kt:51]: Did you look into whether you could use [input chips](https://developer.android.com/develop/ui/compose/components/chip#input) for this ?
At the very least, the margin between the Text and the X icon should be adjusted to match the chip component.

**Review by violoncelloCH — CHANGES_REQUESTED** [review]
Thanks for the incorporation of the tag select screen into the event creation screen! I've just got some minor, mostly design related comments :)

File: app/src/androidTest/java/com/github/swent/echo/compose/event/EventScreenTest.kt [hunk 4]
```diff
@@ -1,47 +1,60 @@
...
+
+    @get:Rule(order = 1) val composeTestRule = createAndroidComposeRule<MainActivity>()
 
     private val mockedRepository = mockk<Repository>(relaxed = true)
     private val mockedAuthenticationService = mockk<AuthenticationService>(relaxed = true)
     private lateinit var eventViewModel: EventViewModel
     private val savedEventId = SavedStateHandle(mapOf())
 
+    // @Inject var tagViewModel: TagViewModel = TagViewModel(mockedRepository)
+
```
[the remark this answers is not in the document]
Comment (unglazedstamp) [reply app/src/androidTest/java/com/github/swent/echo/compose/event/EventScreenTest.kt:49]: Yes, thanks

File: app/src/main/java/com/github/swent/echo/compose/event/EventTagEntry.kt (hunk 1 above)
Comment (unglazedstamp) [reply app/src/main/java/com/github/swent/echo/compose/event/EventTagEntry.kt:59]: That's the role of the `contentDescription`, but it's true that the icon description isn't very descriptive

File: app/src/main/java/com/github/swent/echo/compose/event/EventTagEntry.kt (hunk 2 above)
Comment (unglazedstamp) [reply app/src/main/java/com/github/swent/echo/compose/event/EventTagEntry.kt:39]: I was afraid that too long or too many tags could fill the UI, but after trying it, I see that it makes more sense to have multiple rows.

File: app/src/main/java/com/github/swent/echo/compose/event/EventTagEntry.kt (hunk 3 above)
Comment (unglazedstamp) [reply app/src/main/java/com/github/swent/echo/compose/event/EventTagEntry.kt:51]: Thank you for finding input chips, it's exactly the use case I need for the tags.

**Review by violoncelloCH — APPROVED** [review]

**Pull request merged**